src/aliasthis.c

```diff
@@ -21,18 +21,371 @@
 #include "mtype.h"
 #include "declaration.h"
 #include "tokens.h"
+#include "enum.h"
+#include "template.h"
+#include "arraytypes.h"
 
-Expression *resolveAliasThis(Scope *sc, Expression *e)
+typedef Array<const char *> Strings;
+
+/**
+ * iterateAliasThis resolves alias this subtypes for `e` and applies it to `dg`.
+ * dg should return true, if appropriate subtype has been found. Otherwise it should returns 0.
+ * Also dg can get context through secong parameter `ctx`, which was passed through fourth argument of iterateAliasThis.
+ * `dg` can return result expression through `outexpr` parameter, and if it is not NULL, it will be pushed to `ret` array.
+ * At the first stage iterateAliasThis iterates direct "alias this"-es and pushes non-null `outexpr`-es to `ret`.
+ * If `dg` for one of direct "alias this"-es returns true then `iterateAliasThis` breaks at this stage and returns true through return
+ * value and returned by `dg` expression array through `ret`.
+ * Even if true returned by first `dg` call, remaining direct "alias this"-es will be processed.
+ *
+ * If neither of the direct aliases did not return true iterateAliasThis is recursive applied to direct aliases and base classes and interfaces.
+ * If one of those `iterateAliasThis` calls returns true our `iterateAliasThis` will return true. Otherwise it will return 0.
+ *
+ * The the last argument needs for internal use and should be NULL in user call.
+ */
+
+bool iterateAliasThis(Scope *sc, Expression *e, bool (*dg)(Scope *sc, Expression *aliasexpr, void *ctx, Expression **outexpr), void *ctx, Expressions *ret, bool gagerrors, StringTable *directtypes)
```

Comment:
I suggest pulling out the function pointer type into a well-named typedef, since magicport is going to throw away your nice line-splitting work otherwise.